### DISCUSSION TITLE
Confused about the state of `cb` and system-clipboard

### DISCUSSION THREAD

**Discussion by StrictlyMonad** [Slackadays/Clipboard#117] [Q&A] [answered]
Alright folks, hate to break it to you but apparently I am not understanding how to get `cb` to work with my system-clipboard.

From the limited `man cb` and a bit of browsing in the online man-page (git repo), I have found that there is no way to get `cb` to paste things through `C-v` and also recognize the contents copied through `C-c`. Or am I mistaken?

Note: using the latest released version of `cb` in Nix + xmonad, therefore things might be a tad different that what y'all are experiencing.

**Comment by Slackadays** [comment]
You're right, there's no way to paste things directly through `C-v` nor copy them directly with `C-p` (or is it `C-c`?). However, if you're using X11 or Wayland, then it should work with either of those systems if you use cb directly in the terminal. This also means you either have to use it in the terminal or set up a global shortcut to paste the contents by piping them out to wherever you want on the desktop.

**Reply by StrictlyMonad** [reply]
My bad, wrote `C-p` because I was intending to do something in my WM.. lmao.

So if I understood your message right, `cb copy file.extension` should copy the file to system clipboard if it happens to be an `x11` or a `wayland` system, right? If that is the case, then I wonder what is preventing my `x11` window manager from recognizing content copied with the help of `cb`.

**Reply by Slackadays** [reply]
It could be a xmonad issue or a Nix one because of the isolation features. However, if you want to get closer to the root cause of the problem, then you'll need to compile cb yourself with the necessary x11 libraries.

**Reply by StrictlyMonad** [reply]
Interesting. I guess it's time to debug and PR solution to upstream.

**Comment by StrictlyMonad** [comment]
Let me understand this properly so that I can help resolve the issue. What more is required for `clipboard-jh` (nixos) to build properly?

Compiling the package from master does not return any complaints besides the following warnings:
```
CMake Warning:
  Manually-specified variables were not used by the project:

    BUILD_TESTING
    CMAKE_EXPORT_NO_PACKAGE_REGISTRY
    CMAKE_INSTALL_BINDIR
    CMAKE_INSTALL_DOCDIR
    CMAKE_INSTALL_INCLUDEDIR
    CMAKE_INSTALL_INFODIR
    CMAKE_INSTALL_LIBEXECDIR
    CMAKE_INSTALL_LOCALEDIR
    CMAKE_INSTALL_MANDIR
    CMAKE_INSTALL_OLDINCLUDEDIR
    CMAKE_INSTALL_SBINDIR
    CMAKE_POLICY_DEFAULT_CMP0025
    INSTALL_PREFIX
``` 

Nix-package: https://github.com/wegank/nixpkgs/blob/master/pkgs/tools/misc/clipboard-jh/default.nix

**Reply by Slackadays** [reply]
This looks good, as if that's all that's unused, then there's nothing necessary that's unavailable. Therefore, if you're able to compile CB with X11 support and install it, then the issue is with X11 at runtime. You'll need to set `CMAKE_BUILD_TYPE` to `Debug` to add the debug messages so that you can properly diagnose the issue, so `-DCMAKE_BUILD_TYPE=Debug`. Once you add that, install CB and look out for the messages that mention "looking for libcbx11.so...` and see what you can find from there.

**Reply by StrictlyMonad** [reply]
The only mentions of `libcbx11.so` in the build is:
```
[ 44%] Linking CXX shared module ../../libcbx11.so
In file included from /build/source/src/cb/src/actions/edit.cpp:15:
/build/source/src/cb/src/actions/../clipboard.hpp:223:20: warning: 'std::string JSONescape(const std::string_view&)' defined but not used [8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-function-Wunused-function8;;]
  223 | static std::string JSONescape(const std::string_view& input) {
      |                    ^~~~~~~~~~

354:-- Installing: /nix/store/4b00liq2cf11jrwydhfwiqaj1li6ndjd-clipboard-jh-b038a08e092e07028712cb83a3747777f4b22f3d/lib/libcbx11.so

363:shrinking /nix/store/4b00liq2cf11jrwydhfwiqaj1li6ndjd-clipboard-jh-b038a08e092e07028712cb83a3747777f4b22f3d/lib/libcbx11.so
```

**Reply by Slackadays** [reply]
The mentions of libcbx11.so will be during runtime as that's where it gets loaded, so you'll need to check there first (the build isn't relevant here because we know for a fact it builds on nix)

**Reply by StrictlyMonad** [reply]
Apparently the nix package for `cb` was unable to detect the libs and this was discussed in https://github.com/NixOS/nixpkgs/issues/229659. Now, we just need to submit the change to nixpkgs.

**Comment by heinwol** [comment, marked as answer]
Fixed with https://github.com/NixOS/nixpkgs/pull/233262

**Reply by StrictlyMonad** [reply]
Much appreciated!
